Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java

```diff
@@ -3,54 +3,46 @@
 
 package com.microsoft.maps.kml;
 
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import android.util.Xml;
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
 import com.microsoft.maps.MapElementLayer;
-import com.microsoft.maps.MapIcon;
-import com.microsoft.maps.MapPolygon;
-import com.microsoft.maps.MapPolyline;
-import com.microsoft.maps.modulestools.MapFactories;
+import com.microsoft.maps.moduletools.DefaultMapFactories;
+import com.microsoft.maps.moduletools.MapFactories;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
 
 /**
  * Class that parses KML and returns a new MapElementLayer containing all the shapes outlined in the
- * GeoJSON.
+ * KML.
  *
  * <p>Created by Elizabeth Bartusiak (t-elbart) on 07/21/2020
  */
 public class KMLParser {
 
   private MapElementLayer mLayer;
+  private MapFactories mFactory;
+  private XmlPullParser mParser = Xml.newPullParser();
```

Comment:
Can be final

Reply:
It looks like the two member variables above this can be as well if you pass them to a constructor before calling internalParse.

Reply:
mParser is now final, but mLayer and mFactory may be better as non-final, since they are set in internalParse for testing purposes. The constructor is called in parse() which then calls internalParse(), but the tests make a direct call to internalParse.

Reply:
Resolved in Teams. mLayer and mFactory are now final.